src/components/icon/icon.registry.ts

```diff
@@ -28,44 +43,52 @@ export class IconsRegistry {
     this.listeners.delete(callback);
   }
 
-  private parseSVG(svgString: string): ParsedIcon {
-    const parsed = this._parser.parseFromString(svgString, 'image/svg+xml');
-    const svg = parsed.querySelector('svg');
-
-    if (parsed.querySelector('parsererror') || !svg) {
-      throw new Error('SVG element not found or malformed SVG string.');
+  public setRefsByTheme(theme: Theme) {
+    if (this.theme !== theme) {
+      this.theme = theme;
+
+      for (const { alias, target } of iconReferences) {
+        const external = this.references
+          .get(alias.collection)
+          ?.get(alias.name)?.external;
+
+        this.setIconRef({
+          alias,
+          target: target.get(this.theme) ?? target.get('default')!,
+          overwrite: !external,
+        });
+      }
     }
+  }
 
-    svg.setAttribute('fit', '');
-    svg.setAttribute('preserveAspectRatio', 'xMidYMid meet');
+  public setIconRef(options: IconReferencePair) {
+    const { alias, target, overwrite } = options;
+    const reference = this.references.getOrCreate(alias.collection);
 
-    return {
-      svg: svg.outerHTML,
-      title: svg.querySelector('title')?.textContent ?? '',
-    };
+    if (overwrite) {
+      reference.set(alias.name, { ...target });
+    }
+
+    this.notifyAll(alias.name, alias.collection);
   }
 
-  public register(name: string, iconText: string, collection = 'default') {
-    const namespace = this.getOrCreateCollection(collection);
-    namespace[name] = this.parseSVG(iconText);
+  public getIconRef(name: string, collection: string): IconMeta {
```

Comment:
Just as a note: Could consider eventually exposing this to allow user code to interrogate the registry state if needed (think we've did so in Angular already), but can wait for actual usage to ramp up.